File: app/src/main/java/app/opass/ccip/adapter/ScheduleTabAdapter.java
package app.opass.ccip.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import app.opass.ccip.fragment.ScheduleFragment;

import java.util.ArrayList;
import java.util.List;

public class ScheduleTabAdapter extends FragmentStatePagerAdapter {
    private final List<Fragment> mFragmentList = new ArrayList<>();
    private final List<String> mFragmentTitleList = new ArrayList<>();

    public ScheduleTabAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mFragmentTitleList.get(position);
    }

    public void addFragment(Fragment fragment, String title) {
        mFragmentList.add(fragment);
        mFragmentTitleList.add(title);
    }

    public void toggleStarFilter(boolean isStar) {
        for (Fragment fragment : mFragmentList) {
            ((ScheduleFragment) fragment).toggleStarFilter(isStar);
        }
    }
}

File: app/src/main/java/app/opass/ccip/adapter/ScheduleTabAdapter.kt
package app.opass.ccip.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import app.opass.ccip.fragment.ScheduleFragment
import java.util.*

class ScheduleTabAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
    private val mFragmentList = ArrayList<Fragment>()
    private val mFragmentTitleList = ArrayList<String>()

    override fun getItem(position: Int): Fragment {
        return mFragmentList[position]
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mFragmentTitleList[position]
    }

    fun addFragment(fragment: Fragment, title: String) {
        mFragmentList.add(fragment)
        mFragmentTitleList.add(title)
    }

    fun toggleStarFilter(isStar: Boolean) {
        for (fragment in mFragmentList) {
            (fragment as ScheduleFragment).toggleStarFilter(isStar)
        }
    }
}
